### PR TITLE
extract new relic integration into a plugin

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/samson 

 - hide newrelic config if api is disabled
 - add tests for nested attribute assignment

### Risks
- Low: newrelic related things failing
- Med: newrelic integration getting disabled because nobody has PLUGIN=new_relic set ... (all,-xxx) still works ...

### CODE REVIEW

File: app/controllers/concerns/stage_permitted_params.rb [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
-        :notify_email_address,
-        :deploy_on_release,
-        :datadog_tags,
-        :datadog_monitor_ids,
-        :update_github_pull_requests,
-        :email_committers_on_automated_deploy_failure,
-        :static_emails_on_automated_deploy_failure,
-        :use_github_deployment_api,
-        :no_code_deployed,
+    @stage_permitted_params ||= ([
```
Comment (pschambacher) [inline app/controllers/concerns/stage_permitted_params.rb]: `%i` ?

**Comment by pschambacher** [comment]
👍

**Comment by grosser** [comment]
turns out this new relic stuff never worked because the URL is just wrong ... 

```
Request URL:http://localhost:3000/projects/example-project/stages/production/new_relic.json?initial=true
Request Method:GET
Status Code:404 Not Found
```

**Comment by grosser** [comment]
codeclimate is complaining about all the things that got moved ...

**Pull request merged**